src/main/java/org/ecocean/servlet/DecisionStore.java

```diff
@@ -69,6 +70,7 @@ public void doPost(final HttpServletRequest request, final HttpServletResponse r
                 Decision dec = new Decision(user, enc, key, val);
                 myShepherd.getPM().makePersistent(dec);
                 ids.put(dec.getId());
+                Decision.updateEncounterStateBasedOnDecision(myShepherd, enc);
```

Comment:
i _might_ argue that this `updateEncounterState...()` could be moved to _outside_ (below) the `for ( ... multiple.keySet) {` loop.  maybe.  haha.   i am fine with it being here for multiple reasons (not the least of which is that i am _not sure_ if multiple-mode _is even used any more!_)  but just making a note of it here, in case later there seems to be some kind of slowdown for this being calculated once per loop... when i think it might be sufficient to alter encounter.state _after_ all decisions have been recorded.   note to future hopefully-no-one!!